### PR TITLE
Allow mobile driver extensions to client

### PR DESCRIPTION
**Pull request by sebady**
- [X] By placing an `X` in the preceding checkbox, I verify that I am a PayPal employee or
I have signed the [Contributor License Agreement](https://github.com/paypal/SeLion#contributing)

---

Changes to SeLion client to move specific mobile
driver dependent webdriver implementation into providers
that are external jars that are  service loaded by
SeLion client.

Add new interface for MobileProviders.
Created new modules for Appium, iosdriver and selendroid
specified providers.

### CODE REVIEW

File: selendroid-provider/src/main/java/com/paypal/selion/selendroid/platform/grid/SeLionSelendroidDriver.java [hunk 1]
```diff
@@ -13,18 +13,13 @@
...
 
 import java.net.URL;
 
-import org.openqa.selenium.By;
-import org.openqa.selenium.Capabilities;
-import org.openqa.selenium.Dimension;
-import org.openqa.selenium.Point;
-import org.openqa.selenium.WebDriverException;
-import org.openqa.selenium.WebElement;
+import org.openqa.selenium.*;
```
Comment (mach6) [inline selendroid-provider/src/main/java/com/paypal/selion/selendroid/platform/grid/SeLionSelendroidDriver.java]: import change here. intellij! :\

File: selendroid-provider/src/main/java/com/paypal/selion/selendroid/platform/grid/SeLionSelendroidDriver.java [hunk 2]
```diff
@@ -344,4 +339,15 @@ private void performLongClickAction(Point point) {
     private void performSwipeAction(Point start, Point end) {
         new TouchActions(this).down(start.getX(), start.getY()).move(end.getX(), end.getY()).up(end.getX(), end.getY()).perform();
     }
+
+    @Override
+    public void rotate(DeviceRotation deviceRotation) {
+        //TODO
```
Comment (mach6) [inline selendroid-provider/src/main/java/com/paypal/selion/selendroid/platform/grid/SeLionSelendroidDriver.java:351]: What is TODO? Do we need the empty method? Is this actually an `Override`?

File: selendroid-provider/src/main/java/com/paypal/selion/selendroid/platform/grid/SeLionSelendroidDriver.java (hunk 1 above)
Comment (sebady) [reply selendroid-provider/src/main/java/com/paypal/selion/selendroid/platform/grid/SeLionSelendroidDriver.java]: I will fix this.  Intellij being helpful.

File: selendroid-provider/src/main/java/com/paypal/selion/selendroid/platform/grid/SeLionSelendroidDriver.java (hunk 2 above)
Comment (sebady) [reply selendroid-provider/src/main/java/com/paypal/selion/selendroid/platform/grid/SeLionSelendroidDriver.java:351]: SelendroidDriver implements Rotatable interface which added these two new methods in sel3.  We'll have t o either implement them here or have the next version of selendroiddriver (> 0.17) implements them when it supports sel3.

**Comment by mach6** [comment]
rebased and pushed to branch `sebady-patch-sebady-mobiledrivers` on the parent repo.

**Comment by mach6** [comment]
updated and merged to `develop`

**Pull request closed without merging**